apache/trafficcontrol
# Add Traffic Ops Golang Enforcing Capabilities

#### What does this PR do?

Adds Traffic Ops Golang Enforcing Capabilities

#### Which TC components are affected by this PR?


#### What is the best way to verify this PR?

`go test`; 

1. install TO
2. verify endpoints are accessible to an admin user as before (db migration should be automatic)
3. delete a capability from a role in the database, e.g. `delete from role_capability where role_id = (select id from role where name = 'admin') and cap_name = 'delivery-services-read';`
4. verify the endpoint is no longer accessible, e.g. that `/api/1.2/deliveryservices` returns Unauthorized.
5. Verify servers from an admin user have a visible `iloPassword` field which is not all asterisks.
5. Remove the `servers-read-secure` capability from a user, e.g. `delete from role_capability where role_id = 4 and cap_name = 'servers-read-secure';`
6. Verify servers returned to that user via the API have their `iloPassword` field as asterisks.
7. Create a `secure` Parameter, with a nonempty value
8. Verify that secure parameter's `value` is visible to an Admin user.
9. Delete a user's role's `parameters-read-secure` capability, e.g. `delete from role_capability where role_id = 4 and cap_name = 'parameters-read-secure';`
10. Verify a secure parameter's `value` is all asterisks.

#### Check all that apply
